weatherman.py

```diff
@@ -0,0 +1,167 @@
+"""Weatherman
+This module generates yearly reports on weather data collected from files taken
+from the user provided directory.
+Report no. 1 gives the max/min temperature and max/min humidity of each year.
+Report no. 2 gives the hottest day of each year and the temperature on that day.
+"""
+
+import argparse
+import os
+import sys
+
+
+DEFAULT_MAX = -99
+DEFAULT_MIN = 999
+DEFAULT_STR = 'DATE'
+
+
+class WeatherYearData:
+    """This class is used to set the values required for processing the module.
+    """
+
+    def __init__(self):
+        """All values are initized with global constants. These will be used
+        for multiple comparisons in the module.
+        """
+
+        self.max_temp = DEFAULT_MAX
+        self.min_temp = DEFAULT_MIN
+        self.max_temp_date = DEFAULT_STR
+        self.max_humid = DEFAULT_MAX
+        self.min_humid = DEFAULT_MIN
+
+    def get_data(self, date, max_temp, min_temp, max_humid, min_humid):
+        """Sets values to be printed after comparisons for reports"""
+        if max_temp and max_temp > self.max_temp:
+            self.max_temp = max_temp
+            self.max_temp_date = date
+        if min_temp and min_temp < self.min_temp:
+            self.min_temp = min_temp
+        if max_humid and max_humid > self.max_humid:
+            self.max_humid = max_humid
+        if min_humid and min_humid < self.min_humid:
+            self.min_humid = min_humid
+
+
+class WeatherYearReports:
+    """This class deals with the reports section of the module which includes:
+    recieving data, procesing of data by inherting from the WeatherYearData
+    and printing of relevent reports.
+    """
+
+    def __init__(self, data_dir):
+        """years_data: empty dictionary that takes 'year' as key and all
+        other required data as values.
+        data_dir: directory to collect data from.
+        """
+
+        self.year_data = {}
```

Comment:
`year_data` -> `weather_data`. It is not the data of a single year as the name implies.